packages/mui-material/src/Popper/Popper.tsx

```diff
@@ -0,0 +1,211 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import PopperUnstyled, { PopperUnstyledProps } from '@mui/base/PopperUnstyled';
+import { chainPropTypes, HTMLElementType, refType } from '@mui/utils';
+import { Direction, useThemeWithoutDefault as useTheme } from '@mui/system';
+
+export type PopperProps = Omit<PopperUnstyledProps, 'direction'>;
+
+/**
+ *
+ * Demos:
+ *
+ * - [Autocomplete](https://mui.com/components/autocomplete/)
+ * - [Menus](https://mui.com/components/menus/)
+ * - [Popper](https://mui.com/components/popper/)
+ *
+ * API:
+ *
+ * - [Popper API](https://mui.com/api/popper/)
+ */
+const Popper = React.forwardRef(function Popper(
+  props: PopperProps,
+  ref: React.ForwardedRef<HTMLDivElement>,
+) {
+  const theme = useTheme<{ direction?: Direction }>();
+  return <PopperUnstyled direction={theme?.direction} {...props} ref={ref} />;
+});
+
+function resolveAnchorEl(anchorEl: any) {
+  return typeof anchorEl === 'function' ? anchorEl() : anchorEl;
+}
+
+Popper.propTypes /* remove-proptypes */ = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // |     To update them edit TypeScript types and run "yarn proptypes"  |
+  // ----------------------------------------------------------------------
+  /**
+   * An HTML element, [virtualElement](https://popper.js.org/docs/v2/virtual-elements/),
+   * or a function that returns either.
+   * It's used to set the position of the popper.
+   * The return value will passed as the reference object of the Popper instance.
+   */
+  anchorEl: chainPropTypes(
```

Comment:
Do we need this one here? It seems to be already present in the base, and the logic is tricky to maintain in two different places.

Reply:
By default the generated propTypes look as follows:
```js
{
  anchorEl: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({
      contextElement: function (props, propName) {
        if (props[propName] == null) {
          return null;
        } else if (typeof props[propName] !== 'object' || props[propName].nodeType !== 1) {
          return new Error("Expected prop '" + propName + "' to be of type Element");
        }
      },
      getBoundingClientRect: PropTypes.func.isRequired,
    }),
  ]),
  /**
   * Popper render function or node.
   */
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
    PropTypes.number,
    PropTypes.object,
    PropTypes.shape({
      '__@iterator@428': PropTypes.func.isRequired,
    }),
    PropTypes.string,
    PropTypes.bool,
  ]),
  /**
   * An HTML element or function that returns one.
   * The `container` will have the portal children appended to it.
   *
   * By default, it uses the body of the top-level document object,
   * so it's simply `document.body` most of the time.
   */
  container: PropTypes.oneOfType([
    function (props, propName) {
      if (props[propName] == null) {
        return new Error("Prop '" + propName + "' is required but wasn't specified");
      } else if (typeof props[propName] !== 'object' || props[propName].nodeType !== 1) {
        return new Error("Expected prop '" + propName + "' to be of type Element");
      }
    },
    PropTypes.func,
  ]),
}
```

This causes both ESLint and TS to complain in a couple of places. Plus, even if we solved these problems, it would make the API docs differ between Popper and PopperUnstyled, while they should be the same (sans the `direction` prop).

Reply:
In this case, I propose a

``
```suggestion
  anchorEl: PropTypes /* @typescript-to-proptypes-ignore */.any;
```

Reply:
All right, the PR is ready: #29680